src/web/api_graph.nit

```diff
@@ -0,0 +1,273 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Render various graph from a model.
+module api_graph
+
+import web_base
+import dot
+import uml
+
+# Group all api handlers in one router.
+class APIGraphRouter
+	super Router
+
+	# Model to pass to handlers.
+	var model: Model
+
+	# Mainmodule to pass to handlers.
+	var mainmodule: MModule
+
+	init do
+		use("/inheritance/:id", new APIInheritanceGraph(model, mainmodule))
+	end
+end
+
+# Render a hierarchy graph for `mentity` if any.
+class APIInheritanceGraph
+	super APIHandler
+
+	redef fun get(req, res) do
+		var pdepth = req.int_arg("pdepth")
+		var cdepth = req.int_arg("cdepth")
+		var mentity = mentity_from_uri(req, res)
+		if mentity == null then
+			res.error 404
+			return
+		end
+		var g = new InheritanceGraph(mentity, view)
+		res.send g.draw(pdepth, cdepth).to_svg
+	end
+end
+
+# Graph for mentity hierarchies
+#
+# Recursively build parents and children list from a `center`.
+class InheritanceGraph
+	super ModelVisitor
+
+	# MEntity at the center of this graph
+	var center: MEntity
+
+	# ModelView used to filter graph
+	var view: ModelView
+
+	# Graph generated
+	var graph: DotGraph is lazy do
+		var graph = new DotGraph("package_diagram", "digraph")
+
+		graph["compound"] = "true"
+		graph["rankdir"] = "BT"
+		graph["ranksep"] = 0.3
+		graph["nodesep"] = 0.3
+
+		graph.nodes_attrs["margin"] = 0.1
+		graph.nodes_attrs["width"] = 0
+		graph.nodes_attrs["height"] = 0
+		graph.nodes_attrs["fontsize"] = 10
+		graph.nodes_attrs["fontname"] = "helvetica"
```

Comment:
Could you put just `sans-serif` (like the rest of the UI) or `monospace`, or clear it? I do not know why, but when my web browser sees an unknown font in the `font-family` SVG attribute, it fallbacks to an ugly default.

Here an example with `font-family="Helvetica,sans-Serif"` (what nitweb generates):
![ng-doc_-_2016-06-21_11 46 55](https://cloud.githubusercontent.com/assets/6044484/16236716/573985a0-37a7-11e6-8a08-7fd71cc2823b.png)


If I change it to `font-family="sans-serif"`, `font-family="sans-serif"` or remove the attribute:
![ng-doc_-_2016-06-21_11 52 46](https://cloud.githubusercontent.com/assets/6044484/16236733/6be35d1e-37a7-11e6-8682-f96f68cb00f8.png)

If  I change it to `font-family="monospace"`:
![ng-doc_-_2016-06-21_11 56 35](https://cloud.githubusercontent.com/assets/6044484/16236739/70a616ca-37a7-11e6-9a38-c57170f6532a.png)